### PR TITLE
Add createdAt date time to maps

### LINKED ISSUE #403: Add createdAt date time to maps
**Proposed feature**
We want to show the last maps on the start page with the time at which they were created. However, when a map is created in the backend, the response includes a ServerMap object which includes the following data:

```
export interface IMmpClientMap {
  uuid: string
  lastModified: Date
  deleteAfterDays: number
  deletedAt: Date
  data: IMmpClientNode[]
  options: IMmpClientMapOptions
}
```

This should be enhanced by the created time of the server map. However, I don't think this field is currently available in the database.

- [ ] Create a new `createdAt`field in the database for the map table
- [ ] Export the `createdAt` field on map creation in the endpoint
- [ ] Add the new field to the frontend types and save the field in the storage service within `prepareNewMap` call from map-sync Service

### PR DESCRIPTION
**Pull request by sorenjohanson**
closes #403

### CODE REVIEW

File: teammapper-backend/src/map/entities/mmpMap.entity.ts [hunk 1]
```diff
@@ -36,4 +36,7 @@ export class MmpMap {
   })
   /* eslint-enable @typescript-eslint/no-unused-vars */
   nodes: MmpNode[]
+
+  @Column({ type: 'timestamptz', default: () => 'now()' })
```
Comment (JannikStreek) [inline teammapper-backend/src/map/entities/mmpMap.entity.ts:40]: the default ist not really fitting here. if possible use the modifiedAt as default or leave it null

File: teammapper-backend/src/map/entities/mmpNode.entity.ts [hunk 2]
```diff
@@ -102,4 +102,7 @@ export class MmpNode {
 
   @Column({ type: 'timestamptz', nullable: true, default: () => 'now()' })
   lastModified: Date
+
+  @Column({ type: 'timestamptz', default: () => 'now()' })
```
Comment (JannikStreek) [inline teammapper-backend/src/map/entities/mmpNode.entity.ts:106]: same

**Comment by JannikStreek** [comment]
Please also provide the field to the frontend, currently it's not exposed/returned. Please see issue #403 for details.

File: teammapper-backend/src/map/entities/mmpMap.entity.ts (hunk 1 above)
Comment (sorenjohanson) [reply teammapper-backend/src/map/entities/mmpMap.entity.ts]: I've removed the default here entirely, though I'm guessing that we'll create new maps using `createdAt: new Date()`, so I'm not sure how much of a practical difference that'll make.

File: teammapper-backend/src/map/entities/mmpNode.entity.ts (hunk 2 above)
Comment (sorenjohanson) [reply teammapper-backend/src/map/entities/mmpNode.entity.ts]: Same as above.

File: teammapper-backend/src/map/entities/mmpMap.entity.ts (hunk 1 above)
Comment (JannikStreek) [reply teammapper-backend/src/map/entities/mmpMap.entity.ts]: my primarily concern would be existing maps, where this field should be empty. I think adding a new column with a default value would populate also existing old entries which is not wanted.

Comment (sorenjohanson) [reply teammapper-backend/src/map/entities/mmpMap.entity.ts:40]: Added defaults now in a secondary migration, so existing maps get a null entry in createdAt.

**Pull request merged**